src/dataLoaders/actions/dataLoaders.ts

```diff
@@ -407,12 +409,21 @@ export const createOrUpdateTelegrafConfigAsync = () => async (
   )
 
   if (telegrafConfigID) {
-    const telegraf = await client.telegrafConfigs.update(telegrafConfigID, {
-      name: telegrafConfigName,
-      description: telegrafConfigDescription,
-      plugins,
+    const response = await putTelegraf({
+      telegrafID: telegrafConfigID,
+      data: {
+        name: telegrafConfigName,
+        description: telegrafConfigDescription,
+        // TODO: interface TelegrafPlugin does not have property 'type'
```

Comment:
I don't understand this comment. Can you explain what you mean by this TODO?

The actual TypeScript warning here is `Object literal may only specify known properties, and 'plugins' does not exist in type 'TelegrafRequest'`

Reply:
This TODO is related to the typecasting in another TODO on line # 399.

'plugins' did not exist in type 'TelegrafPluginRequest.' After a [fix](https://github.com/influxdata/openapi/pull/246) in OpenAPI, and an update of SHA in `package.json` for `generate` script, 'plugins' is a valid field now. 

I think I should remove the TODO here because the warning is gone now. I will update the TODO on line # 399 to make it more clear. Just FYI, I wrote an issue #3433 related to this.